overlay/usr/lib/inithooks/bin/opencart.py

```diff
@@ -88,29 +89,25 @@ def php_uniqid(prefix=''):
         return prefix + hex(int(time.time()))[2:10] + hex(int(time.time() * 1000000) % 0x100000)[2:7]
 
     subprocess.run(["sed", "-ri",
-        "s|('HTTP(S?)_SERVER',) '.*'|\\1 'http\\L\\2://%s/'|g" % domain,
+        "s|('HTTP(S?)_SERVER',) '.*'|\\1 'https\\L\\2://%s/'|g" % domain,
         "/var/www/opencart/config.php"])
     subprocess.run(["sed", "-ri",
-        "s|('HTTP(S?)_SERVER',) '.*'|\\1 'http\\L\\2://%s/admin/'|g" % domain, 
-        "/var/www/opencart/admin/config.php"])
+        "s|('HTTP(S?)_SERVER',) '.*'|\\1 'https\\L\\2://%s/turnkey_admin/'|g" % domain, 
+        "/var/www/opencart/turnkey_admin/config.php"])
     subprocess.run(["sed", "-ri",
-        "s|('HTTP(S?)_CATALOG',) '.*'|\\1 'http\\L\\2://%s/'|g" % domain,
-        "/var/www/opencart/admin/config.php"])
-    salt = hashlib.md5(php_uniqid(str(randint(100000000, 999999999))).encode('utf8')).hexdigest()[:9]
+        "s|('HTTP(S?)_CATALOG',) '.*'|\\1 'https\\L\\2://%s/'|g" % domain,
+        "/var/www/opencart/turnkey_admin/config.php"])
 
     apache_conf = "/etc/apache2/sites-available/opencart.conf"
     subprocess.run(["sed", "-i", "\|RewriteRule|s|https://.*|https://%s/\$1 [R,L]|" % domain, apache_conf])
     subprocess.run(["sed", "-i", "\|RewriteCond|s|!^.*|!^%s$|" % domain, apache_conf])
     subprocess.run(["service", "apache2", "restart"])
 
-    password_hash = hashlib.sha1(password.encode('utf8')).hexdigest()
-    password_hash = hashlib.sha1((salt + password_hash).encode('utf8')).hexdigest()
-    password_hash = hashlib.sha1((salt + password_hash).encode('utf8')).hexdigest()
+    password_hash = phpass_context.hash(password)
```

Comment:
TBH I haven't used passlib before, but this seems like a nice clean way to do this!? FWIW Stefan used a PHP snippet in one app recently because it just would not work with anything we tried (it appeared to using the built in PHP bcrypt but it wouldn't work with python-bcrypt - I wonder if this might have been a better/easier way to achieve that?